Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/Lights/Ltc.azsli

```diff
@@ -503,14 +569,88 @@ void LtcPolygonEvaluate(
     // Evaluate all the points
     for (uint curIdx = startIdx + 1; curIdx < endIdx; ++curIdx)
     {
-        float3 p1 = mul(orthonormalMat, positions[curIdx].xyz - pos); // Current point in polygon
-        EvaluatePolyEdge(p0, p1, prevClipPoint, ltcMat, diffuse, specular);
+        float3 p1 = mul(orthonormalMat, positions[curIdx].xyz - surface.position); // Current point in polygon
+        EvaluatePolyEdge(p0, p1, ltcMat, prevClipPoint, diffuse, specular);
         p0 = p1;
     }
     
-    EvaluatePolyEdge(p0, closePoint, prevClipPoint, ltcMat, diffuse, specular);
+    EvaluatePolyEdge(p0, closePoint, ltcMat, prevClipPoint, diffuse, specular);
 
     // Note: negated due to winding order
     diffuse = -diffuse;
     specular = -specular;
+
+    // Apply BRDF scale terms (BRDF magnitude and Schlick Fresnel)
+    float2 schlick = ltcAmpMatrix.Sample(PassSrg::LinearSampler, ltcCoords).xy;
+    float3 specularRgb = specular * ((schlick.x * surface.specularF0) + (1.0 - surface.specularF0) * schlick.y);
+
+    if(o_clearCoat_feature_enabled)
```

Comment:
Same comment, this could be wrapped in a function since a lot of it is similar to code for existing specular.

Reply:
I can wrap some of it up, but there's subtle differences in a lot of it. Like clearcoat only needs a single float specularF0 vs a float3 for standard specularF0 (because of metallics). Clearcoat also needs to transform and clip the polygon separately, but doesn't need to compute diffuse.